_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,82 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The aim of this project is to automatically create whitelists (to be utilized
+by vulture) for projects using sip files to generate Python bindings for C++
+code, most significant of them being PyQt. Once implemented, a whitelist
+generated through this generator for PyQt could be universally utilised across
+all PyQt projects.
+
+#### IDEAS
+
+* Create a command line tool which takes in sip files as input and spits a
+  whitelist with all the virtual functions defined in the sip files. (Add
+  Documentation and tests)
+* Use this tool to generate a whitelist for PyQt and ship it with vulture. This
+  will reduce overhead for PyQt projects which use vulture.
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding
+  generator used for PySide2
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `xml`
+  output from coverage.py
+* Get familiar with coverage.xml
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst)
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/28)
+
+##### CODING PHASE 1
+* The whitelist generator should be completely implemented.
+* It should reside in it's own repository, with CI enabled for testing on both
+  the GNU/Linux and Windows platforms.
+* It should be gracefully documented.
+* A whitelist for PyQt should be generated to be then shipped with vulture by
```

Comment:
@jendrikseipp agrees with bundling the PyQt whitelist into the vulture releases?

That doesnt sound very scalable, as then vulture would be asked to ship other whitelists.  Generation of whitelists feels like a task that vulture users should do, and possibly store in their repo and regenerate periodically when their dependencies change.

Reply:
> @jendrikseipp agrees with bundling the PyQt whitelist into the vulture releases?

I know that he he didn't like this idea initially, but since then, there have been some major advancements in the way vulture handles whitelists. Now, we've whitelists for every library for which we know that vulture raises certain false positives (https://github.com/jendrikseipp/vulture/tree/master/vulture/whitelists). Using this support infrastructure for whitelisting, it would be a lot nicer to ship whitelist with vulture, wouldn't it @jendrikseipp?

> That doesnt sound very scalable, as then vulture would be asked to ship other whitelists. 

Yes, it does requires some efforts for maintaining those, but I guess, owing to how much user friendly it makes vulture, it's worth it. And also, it was also one of my task to create whitelists for popular libraries during last year's GSoC, remember?

> Generation of whitelists feels like a task that vulture users should do, and possibly store in their repo and regenerate periodically when their dependencies change.

Since not all whitelists need to be dynamically generated, it wouldn't be very user friendly to let users do the mundane work of creating a whitelist and including it with vulture.